### PR TITLE
Implement `inccommand`

### PR DESCRIPTION
**Pull request by adrsm108**
**What this PR does / why we need it**:
This PR introduces the option `inccommand`, which enables a live preview of `:s` commands as they are typed. It also makes some improvements to interactive substitution with the `c` flag. 

The feature is configured by setting `vim.inccommand` to `'append'` or `'replace'` (or `''` to disable).
When set to `'append'`, substitution previews are inserted after each highlighted match, a behavior found in some popular vim emulators like spacemacs.
When set to `'replace'`, the substitution previews replace each match, which is consistent with the default behavior of neovim.

The config keys `vim.substitutionColor` and `vim.substitutionTextColor` are provided to control the appearance of preview text. 

**Which issue(s) this PR fixes**
#6656
Also addresses #6655

**Special notes for your reviewer**:
Probably the biggest change from the original neovim behavior is how multi-line substitutions are handled. Since decorations can't change the apparent number of lines in the editor, concealed multi-line matches still have line breaks that won't be present once the substitution is performed. Multi-line replacements also have to be displayed inline; I currently do this by replacing line breaks with the character `⏎` (U+23CE, "RETURN SYMBOL"), but it might be more sensible to use standard escape sequences.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,6 +131,8 @@ These settings are specific to VSCodeVim.
...
 | vim.searchMatchTextColor         | Foreground color of current search match                                                                                                                                                                                                                                                                                                                                                                                            | String  | None                                      |
+| vim.substitutionColor            | Background color of substituted text when `vim.inccommand` is enabled                                                                                                                                                                                                                                                                                                                                                               | String  | "#50f01080"                               |
```
Comment (J-Fields) [inline README.md:134]:
```suggestion
| vim.substitutionColor            | Background color of substitution text when `vim.inccommand` is enabled                                                                                                                                                                                                                                                                                                                                                               | String  | "#50f01080"                               |
```

Maybe? This is the text that is doing the substituting, not being substituted, right?

File: README.md [hunk 2]
```diff
@@ -423,6 +425,7 @@ Configuration settings that have been copied from vim. Vim settings are loaded i
...
 | vim.ignorecase   | Ignore case in search patterns                                                                                                                                                                                                                                | Boolean | true                                                           |
 | vim.incsearch    | Show the next match while entering a search                                                                                                                                                                                                                   | Boolean | true                                                           |
+| vim.inccommand   | Show the results of a `substitute` command while typing.                                                                                                                                                                                                      | String  | `replace`                                                      |
```
Comment (J-Fields) [inline README.md:428]:
```suggestion
| vim.inccommand   | Show the results of the `:substitute` command while typing                                                                                                                                                                                                      | String  | `replace`                                                      |
```

**Comment by adrsm108** [comment]
I'm not super familiar with github actions, but is this check working as intended? The only changes were to the README, and things seem to be going wrong when it's reformatted.

**Review by J-Fields — APPROVED** [review]
Thanks a lot!

**Comment by adrsm108** [comment]
My pleasure!

**Comment by Swoorup** [comment]
I have been using this plugin for quite sometime, and didn't realise this feature had landed until I saw the preview when using the replace command. Just came to say thank you. 🙏 😊 🤩

**Pull request merged**